tools/provision

```diff
@@ -1,33 +1,48 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 BASEDIR=$(dirname `dirname $0`)
+VENVEXECDIR="bin"
+PYVERSION=$(python -V | grep -Po '(?<=Python ).+')
+PYCUSTOMPATH_ARG=""
+while getopts "hp:" arg; do
+    case "$arg" in
+        h)
+            echo "USAGE: $0 [-p <path_to_python_version>]"
+            exit
+            ;;
+        p)
+            PYCUSTOMPATH_ARG="-p $OPTARG"
+            PYVERSION=$("$OPTARG" "-V" 2>&1 | grep -Po '(?<=Python ).+')
+            ;;
+    esac
+done
+VENVDIR="zulip-api-py$PYVERSION-venv"
 
-if [ ! -d "$BASEDIR/zulip-api-py2-venv" ]; then
-    virtualenv $BASEDIR/zulip-api-py2-venv
+if [ ! -d "$BASEDIR/$VENVDIR" ]; then
+    virtualenv $PYCUSTOMPATH_ARG "$BASEDIR/$VENVDIR"
     echo "Virtualenv created."
 fi
 
-VENVBINDIR="bin"
-if [[ "$OSTYPE" == "msys" ]]; then
+if [[ ! -d "$BASEDIR/$VENVDIR/$VENVEXECDIR" ]]; then
```

Comment:
"$OSTYPE" is a variable that doesn't even exist on Windows; `msys` is only set by CygWin (a terminal environment for Windows that emulates some linux behavior).
If we want to be comrehensive with  the variable check approach, we would need to check these cases:
```
        # Windows
if [[ "$OSTYPE" exists ]]; then
        # Mac OSX
elif [[ "$OSTYPE" == "cygwin" ]]; then
        # POSIX compatibility layer and Linux environment emulation for Windows
elif [[ "$OSTYPE" == "msys" ]]; then
        # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
elif [[ "$OSTYPE" == "win32" ]]; then
```
I thought it to be much easier to check on the existence of the `Scripts` directory, since that is the only difference for `venv` between Windows and Linux.